### PR TITLE
chore: publish to crates.io

### PR DESCRIPTION
**Pull request by wischli**
## fixes https://github.com/KILTprotocol/ticket/issues/565
- [x] TODO: Add github action which publishes on new versions

## How to test:
Checkout on [crates.io](https://crates.io/crates/portablegabi-pallet)

## Checklist:

- [x] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Publish
+
+on:
+  release:
+    types: [published, created, edited]
```
Comment (weichweich) [inline .github/workflows/publish.yml:5]: Wouldn't `published` or `created` be enough? I just want to make sure that we don't run this action too often. 😁

File: Cargo.toml [hunk 2]
```diff
@@ -3,6 +3,10 @@ name = "portablegabi-pallet"
 version = "0.3.0"
 authors = ["KILT <info@kilt.io>"]
 edition = "2018"
+repository = "https://github.com/KILTprotocol/portablegabi-pallet"
+homepage = "https://github.com/KILTprotocol/portablegabi"
```
Comment (weichweich) [inline Cargo.toml:7]: I would maybe link to our portablegabi tutorial? But this is also fine.

**Review by weichweich — APPROVED** [review]

File: Cargo.toml (hunk 2 above)
Comment (wischli) [reply Cargo.toml:7]: Agree, will change.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (wischli) [reply .github/workflows/publish.yml:5]: Agree, will change.

**Review by weichweich — APPROVED** [review]

**Pull request merged**